### PR TITLE
Assorted fixes for airlock controller

### PR DESCRIPTION
**Pull request by PsyCommando**
## Description of changes
The airlock controller have been really buggy for a while now, so I've trying fixing the problems I could figure out.

For one, sensors, buttons and dummy controllers wouldn't sync their icon state properly at all, and animations meant to happen during cycling wouldn't always fire properly. Or the indicators on some of the airlock machines wouldn't change their state properly.

* Added proper callback when the master controller is cycling the airlock, so individual machines can keep up.
* Added a way to configure dummy airlock terminals via multitool.
* Made the airlock controller UI a bit less messy, and adjusted the window size so it stops overflowing, and causing scroll bars to appear.
* Fixed the controller id tag multitool extension's sanitizing not working.
* Prevented the controller from spawning duplicate parts on spawn.
* Made some slightly less ugly sprites for the airlock button and sensors, with directional sprites and layered indicator lights and animations.
* Made the airlock stuff work more like most wall mounted machines.
* Added subtle sound for embedded controller interactions. Since it didn't make any before.
* Added several mapper presets for vents, air alarms and etc for making setting up an airlock much less annoying, with all the proper frequencies and filters already set up.
* Added terminal presets for external vents connected via terminal and etc..

## Changelog
:cl:
tweak:  Tweaked the airlock controller UI a little bit for the advanced airlock controller. Makes it fit better in its window, and reorganized the buttons around to look a bit less messy.
tweak: Adjusted directional offsets for airlock devices, so they actually are aligned to walls. 
tweak: Adjusted directional offsets of warning signs since they kept covering up airlock devices on the tradeship map.
bugfix: Fixed sanitizing id_tags set via multitool on the airlock machinery.
bugfix: Fixed airlock machinery generally falling out of sync with eachother.
imageadd: Added new icons for airlock sensor and airlock access button. They have overlays and directions now.
code: Added templates for mappers for air vents, air alarms, airlock stuff. 
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
CI fail appears valid.

**Comment by PsyCommando** [comment]
> CI fail appears valid.

Yeah, it's fixed.
I just realized I had a few things missing.

**Comment by PsyCommando** [comment]
Also, I put the construction state fixes from this PR into #2839

File: nano/templates/advanced_airlock_console.tmpl [hunk 1]
```diff
@@ -33,28 +39,17 @@
...
-			{{:helper.link('Force interior door', 'alert', {'command' : 'force_int'}, null, data.processing ? 'yellowButton' : null)}}
-		</div>
-	</div>
-	<div style="padding-left: 10px; width: 80px; float: left">
-		<div class="item">
-			<div class="itemContent" style="padding-bottom: 2px; width: auto">
-				{{:helper.link('Purge', 'refresh', {'command' : 'purge'}, data.processing ? 'disabled' : null, data.purge ? 'linkOn' : null)}}
-			</div>
-			<div class="itemContent" style="width: auto">
-				{{:helper.link('Secure', data.secure ? 'locked' : 'unlocked', {'command' : 'secure'}, data.processing ? 'disabled' : null, data.secure ? 'linkOn' : null)}}
```
Comment (MistakeNot4892) [inline nano/templates/advanced_airlock_console.tmpl:53]: Did you remove Secure?

File: code/game/objects/structures/signs.dm [hunk 2]
```diff
@@ -86,6 +86,7 @@
 /obj/structure/sign/warning
 	name = "\improper WARNING"
 	icon_state = "securearea"
+	directional_offset = "{'NORTH':{'y':-32}, 'SOUTH':{'y':32}, 'WEST':{'x':34}, 'EAST':{'x':-34}}"
```
Comment (MistakeNot4892) [inline code/game/objects/structures/signs.dm:89]: You already know what I'm going to say about unrelated changes and monolithic PRs.

File: code/game/objects/items/weapons/circuitboards/wall.dm [hunk 3]
```diff
@@ -75,10 +75,11 @@
 
 /obj/item/stock_parts/circuitboard/airlock_controller/get_buildable_types()
 	. = list()
-	for(var/path in typesof(/obj/machinery/embedded_controller/radio))
+	for(var/path in (subtypesof(/obj/machinery/embedded_controller/radio) | subtypesof(/obj/machinery/embedded_controller/radio/airlock)))
```
Comment (MistakeNot4892) [inline code/game/objects/items/weapons/circuitboards/wall.dm]: Would possibly be better to make this a local static list.

File: nano/templates/advanced_airlock_console.tmpl (hunk 1 above)
Comment (PsyCommando) [reply nano/templates/advanced_airlock_console.tmpl:53]: Nah, moved it to the top. Its on the same line as something else so its a bit hard to spot it, but its there.

File: code/game/objects/items/weapons/circuitboards/wall.dm (hunk 3 above)
Comment (PsyCommando) [reply code/game/objects/items/weapons/circuitboards/wall.dm]: Alright, will do. 
But, any reasons why they rarely use static lists in path checking for loops around the code though?

File: code/game/objects/structures/signs.dm (hunk 2 above)
Comment (PsyCommando) [reply code/game/objects/structures/signs.dm:89]: Alright, I'll make another PR with that.
But the main reason this was done is because the signs would cover the airlock controls, so I moved them, and then noticed that the auto-offset for them was off because not all signs seems to be oriented the same way on their sprites.

File: code/game/objects/items/weapons/circuitboards/wall.dm (hunk 3 above)
Comment (PsyCommando) [reply code/game/objects/items/weapons/circuitboards/wall.dm]: Speaking of that, maybe writing a cached subtypesof proc be a good idea? Since subtypes won't really change at runtime and all.

Comment (PsyCommando) [reply code/game/objects/items/weapons/circuitboards/wall.dm]: Also done.

File: code/game/objects/structures/signs.dm (hunk 2 above)
Comment (PsyCommando) [reply code/game/objects/structures/signs.dm:89]: https://github.com/NebulaSS13/Nebula/pull/2843

**Review by MistakeNot4892 — APPROVED** [review]
Honestly not terribly confident that I spotted all possible issues but heigh ho here we go.
